process-dataset/index.js

```diff
@@ -1,90 +1,119 @@
-const uploadFeatureDefs = require("./steps/upload-feature-defs");
-const uploadCellLines = require("./steps/upload-cell-lines");
-const formatAndWritePerCellJsons = require("./steps/write-per-cell-jsons");
-const uploadCellCountsPerCellLine = require("./steps/upload-cell-counts");
-const uploadFileInfo = require("./steps/upload-file-info");
-const uploadFileToS3 = require("./steps/upload-to-aws");
-const uploadDatasetImage = require("./steps/upload-dataset-image")
+const dataPrep = require("./data-validation/data-prep");
+const schemas = require("./data-validation/schema");
+const processSingleDataset = require("./process-single-dataset");
 
-const FirebaseHandler = require('../firebase/firebase-handler');
-
-const TEMP_FOLDER = "./data";
 const args = process.argv.slice(2);
 console.log('Received: ', args);
 
-const datasetReadFolder = args[0];
-const skipFileInfoUpload = args[1];
+const inputFolder = args[0];
+const shouldSkipFileInfoUpload = args[1] === "true";
 
-const readDatasetInfo = async () => {
-    const data = await fsPromises.readFile(`${datasetReadFolder}/dataset.json`)
+const readDatasetJson = async (folder) => {
+    const data = await fsPromises.readFile(`${folder}/dataset.json`)
     return JSON.parse(data)
 }
 
-const processDataset = async () => {
+// Extract out just the fields that will go into Firebase under 
+// dataset-descriptions/{megaset name}
+const getDatasetInfo = (datasetJson) => {
+    const datasetInfo = dataPrep.initialize(datasetJson, schemas.datasetSchema)
+    datasetInfo.production = false; // by default upload all datasets as a staging set
+    return datasetInfo;
+}
+
+const getDatasetId = (dataset) => {
+    return `${dataset.name}_v${dataset.version}`;
+}
 
-    if (!datasetReadFolder) {
+const processMegaset = async () => {
+    if (!inputFolder) {
         console.log("NEED A DATASET FOLDER TO PROCESS")
         process.exit(1)
     }
     
-    fsPromises.readdir(datasetReadFolder)
+    fsPromises.readdir(inputFolder)
         .catch ((error) => {
             console.log("COULDN'T READ DIRECTORY:", error)
-        })
+        }) 
     
-    const datasetJson = await readDatasetInfo()
-    const {
-        name,
-        version,
-    } = datasetJson;
-    const firebaseHandler = new FirebaseHandler(name, version);
-    console.log("Dataset id:", firebaseHandler.id)
-    const fileNames = {
-        featureDefs: datasetJson.featureDefsPath,
-        featuresData: datasetJson.featuresDataPath,
-        cellLineData: datasetJson.cellLineDataPath,
-    }
-    for (const key in fileNames) {
-        if (Object.hasOwnProperty.call(fileNames, key)) {
-            if (!fileNames[key]) {
-                console.error("Missing file name:", key);
-                process.exit(1);
-            }
+    // Top-level megaset structure for Firebase
+    let megasetInfo = {
+        title: "",
+        name: "",
+        publications: [],
+        datasets: {},
```

Comment:
this is an array in the sample json i saw above, but an object here.  This is one of those places where typescript would be nice to really declare/document some schema more explicitly.  Or json schema if you want to be heavy handed.

Reply:
Yeah, I agree it's confusing and TS would be great. I guess we thought in the user provided json it makes more sense to just provide an array of sub-dataset names as strings? But while working with Firebase we thought it was better to have objects with dataset ids as keys.... @meganrm do you think we should change anything?

Reply:
that makes sense to me.. it just wasn't clear at first glance that it's being transformed from what is in the file, since the property names are the same.

Reply:
Okay, I added a comment, so hopefully that makes it more clear!